PolicyServicePkg/PolicyService/PolicyCommon.h

```diff
@@ -9,15 +9,44 @@
 #ifndef _POLICY_COMMON_H_
 #define _POLICY_COMMON_H_
 
+#pragma pack(1)
+
 typedef struct _POLICY_HOB_HEADER {
   EFI_GUID    PolicyGuid;
   UINT64      Attributes;
   UINT16      PolicySize;
   UINT16      AllocationSize;
-  UINT32      Removed  : 1;
-  UINT32      Reserved : 31;
+  UINT16      Removed  : 1;
+  UINT16      Reserved : 15;
+  UINT16      NameSize;
+  // CHAR16 Name[NameSize]

```

Comment:
Does it work if we leave it as `CHAR16 Name[];`?

Reply:
It's variable length. So it would need at least length 1 since the compilers do not like 0-length arrays. I'd rather just leave it as an appended field personally.

Reply:
Sure. I think I may actuall keep the macro though because the easy NULL check is good and will prevent accidental misuse of the array.